### PR TITLE
add ultravnc single click prompt

### PR DESCRIPTION
**Pull request by jianye-dev**

### CODE REVIEW

File: rfbproto.rst [hunk 1]
```diff
@@ -374,6 +374,7 @@ Number      Name
 2           `VNC Authentication`_
 16          `Tight Security Type`_
 19          `VeNCrypt`_
+107         `UltraVNC Single Click Prompt`_
```
Comment (CendioOssman) [inline rfbproto.rst]: Hmmm.... I'm looking at the IANA link and this number isn't registered. If UltraVNC are using it without any other handshake then they really need to register it. Do they have other authentication numbers they use? We need to have this allocated with IANA before we can start recommending it to more implementations.

File: rfbproto.rst [hunk 2]
```diff
@@ -419,6 +420,7 @@ handshaking was successful.
 No. of bytes    Type    [Value]     Description
 =============== ======= =========== ===================================
 4               ``U32``             status:
+..                      -1          restart
```
Comment (CendioOssman) [inline rfbproto.rst]: The status is unsigned, so for clarity I think it is better to write this as `0xffffffff`.

Comment (CendioOssman) [inline rfbproto.rst]: We don't really have an allocation policy for these. I think we need to mail IANA and see if they can set up a registry. We don't want to have conflicts here.

Comment (CendioOssman) [inline rfbproto.rst]: How is this restart defined? What should a client do when it sees this result?

File: rfbproto.rst [hunk 3]
```diff
@@ -783,6 +785,28 @@ Authentication continues with the SASL method when TLS handshake is completed.
...
+UltraVNC Single Click Prompt
+----------------------------
+
+UltraVNC SC Prompt authentication is to be used and protocol data is to be sent
+unencrypted. The server sends a computer information:
+
+========================== ============= ==========================
+No. of bytes               Type          Description
+========================== ============= ==========================
+4                          ``U32``       *prompt-length*
```
Comment (CendioOssman) [inline rfbproto.rst]: Didn't you state that this is in little-endian in your PR to noVNC?

File: rfbproto.rst [hunk 4]
```diff
@@ -783,6 +785,28 @@ Authentication continues with the SASL method when TLS handshake is completed.
...
+*prompt-length*            ``U8`` array  *prompt-string*
```
Comment (CendioOssman) [inline rfbproto.rst]: And what is in this? What should the client do with it?

File: rfbproto.rst [hunk 5]
```diff
@@ -783,6 +785,28 @@ Authentication continues with the SASL method when TLS handshake is completed.
...
+========================== ============= ==========================
+
+The client sends a non-zero value to accept.
```
Comment (CendioOssman) [inline rfbproto.rst]: Why does the client need to accept something?

**Review by CendioOssman — CHANGES_REQUESTED** [review]
Thanks for your contribution. There is unfortunately still not quite clear how to use this extension. Please have a look at the comments and see if you can make it more obvious for someone implementing a client or server.

File: rfbproto.rst (hunk 2 above)
Comment (secureinfo) [reply rfbproto.rst]: The server will resend some new authentication types to be selected by the client and then continue the authentication process.

File: rfbproto.rst (hunk 1 above)
Comment (CendioOssman) [reply rfbproto.rst]: Let's start with the allocations and deal with the behaviour separately. Can you mail IANA and get these numbers allocated? A quick glance suggests that UltraVNC has laid claim to the entire `0x68` to `0x7f` range.

File: rfbproto.rst (hunk 2 above)
Comment (CendioOssman) [reply rfbproto.rst]: You don't need to be a UltraVNC developer in these cases. Just tell IANA that you've found that UltraVNC uses these numbers and that they should be noted in their lists. I've done that in the past without issue. :)

Comment (CendioOssman) [reply rfbproto.rst]: The question is if this is really a "restart", or a "continue". I.e. should the client throw away any information already agreed upon? Or is it a request for the server for several steps?

**Comment by CendioOssman** [comment]
@secureinfo, are you still looking at getting this completed?

**Comment by CendioOssman** [comment]
No response. Closing.

**Comment by Neustradamus** [comment]
@RudiDeVos: Have you seen this PR and comments?

**Pull request closed without merging**